# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/thebes/gfxFontEntry.cpp
# Commit: 8b26219a4294
# Full Hash: 8b26219a4294a083f2d22da23bd8958234e4dd1b
# Author: Jonathan Kew <jkew@mozilla.com>
# Date: 2022-08-16 16:01:29
# Regressor Bug: 1740530
# File Overlap Count: 1
# Description:
#   Bug 1740530 - patch 1 - Move COLR font support functions from gfxUtils into their own class. r=gfx-reviewers,lsalzman
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D152036
# ==============================================================================

diff -r 8d99a5abccc7 -r 8b26219a4294 gfx/thebes/gfxFontEntry.cpp
--- a/gfx/thebes/gfxFontEntry.cpp	Tue Aug 16 06:00:43 2022 -0400
+++ b/gfx/thebes/gfxFontEntry.cpp	Tue Aug 16 10:39:50 2022 +0000
@@ -389,7 +389,7 @@
   auto* colr = GetFontTable(TRUETYPE_TAG('C', 'O', 'L', 'R'));
   auto* cpal = colr ? GetFontTable(TRUETYPE_TAG('C', 'P', 'A', 'L')) : nullptr;
 
-  if (colr && cpal && gfxFontUtils::ValidateColorGlyphs(colr, cpal)) {
+  if (colr && cpal && gfx::COLRFonts::ValidateColorGlyphs(colr, cpal)) {
     if (!mCOLR.compareExchange(nullptr, colr)) {
       hb_blob_destroy(colr);
     }
@@ -1043,9 +1043,9 @@
     uint32_t aGlyphId, const mozilla::gfx::DeviceColor& aDefaultColor,
     nsTArray<uint16_t>& aLayerGlyphs,
     nsTArray<mozilla::gfx::DeviceColor>& aLayerColors) {
-  return gfxFontUtils::GetColorGlyphLayers(GetCOLR(), GetCPAL(), aGlyphId,
-                                           aDefaultColor, aLayerGlyphs,
-                                           aLayerColors);
+  return gfx::COLRFonts::GetColorGlyphLayers(GetCOLR(), GetCPAL(), aGlyphId,
+                                             aDefaultColor, aLayerGlyphs,
+                                             aLayerColors);
 }
 
 typedef struct {